### PR TITLE
Modernization

### PR DESCRIPTION
**Pull request by akx**
* Replace JSHint with Eslint with semistandard style
  * Fix its kvetches
* Run some codemods (all code should still be compatible with Node 6 according to node.green)

### CODE REVIEW

**Comment by rzr** [comment]
Thanks for this patch serie I would prefer for next time to have them submitted in smaller batch (1 per PR would be nice too) then we can track where it break for linux...

I didnt check yet but is it easy to fix CI errors ?

**Comment by akx** [comment]
> I didnt check yet but is it easy to fix CI errors ?

Yeah, they are. Lint warnings, I swear I fixed these but...

**Comment by rzr** [comment]
ok good luck
I will release a new version then.

Would you be interested to co-maintain it ?

See procedure at:
https://abandonware.github.io/

File: .github/workflows/nodepackage.yml [hunk 1]
```diff
@@ -46,3 +46,10 @@ jobs:
           fi
           npm install
           npm test
+      - name: Lint
+        run: |
+          if [ ${{ matrix.node }} = 6 ]; then
```
Comment (rzr) [inline .github/workflows/nodepackage.yml]: I guess lower versions are not working maybe use "-le" instead of "="

Comment (akx) [reply .github/workflows/nodepackage.yml]: The engines stanza in package.json already says the package only supports Node 6+.

File: test/test-service.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-var should = require('should');
+require('should');
```
Comment (rzr) [inline test/test-service.js]: how can this module be used if not named ?
maybe it should not be imported in the first place ?

Comment (akx) [reply test/test-service.js]: [It hooks up `.should.` stuff globally on all objects.](https://github.com/shouldjs/should.js/blob/690405e879d2a46c3bac38fbfabe1dce3c7de44d/should.js#L4349-L4361)

**Comment by akx** [comment]
@rzr Rebased.

**Comment by rzr** [comment]
I think it's desirable but I let it open for other @abandonware/reviewers 

Ping me in a couple of weeks if it isn't merged


May I suggest to update commit message to add references to origin of this patch :

https://github.com/abandonware/noble/pull/38

and upstream too

https://www.slideshare.net/rzrfreefr/updownstreamflows20190411rzr

Thanks again,

**Comment by akx** [comment]
> May I suggest to update commit message to add references to origin of this patch :
> 
> #38

I'm not sure I follow. If this is merged with a regular merge commit (as is usually desirable with larger patch sets like this), the merge commit will have a reference to the PR.

**Comment by rzr** [comment]
It's not mandatory just a suggestion

well if rebased if will be gone, 
and IMHO, adding a depencency to github is not future proof.

**Comment by akx** [comment]
> well if rebased if will be gone,

People generally don't rebase master though :)

> and IMHO, adding a depencency to github is not future proof.

An addition to the commit messages referring to Github would also be a dependency on Github...

**Comment by rzr** [comment]
Please consider about forwarding patches to upstream
Also add cross references in commit messages and I merge this once done and release a new version.

Then in future I plan to rebase on upstream....

**Comment by akx** [comment]
> Please consider about forwarding patches to upstream

Last activity in any upstream Noble packages has been in 2018. Beyond Github, I have sent the upstream folks several mails both via regular mail and their Open Collective (they're apparently getting sponsor money monthly...) to no avail. I'll certainly begin upstreaming patches if they show any sign of life, but at this point I'd rather not waste more time on that.

**Comment by rzr** [comment]
Yes I know that's the main reason I created this org
may this be reported to @OpenCollective to see how can this be handled better
anyway I'll keep updated PR of relevent changes:

Relate-to: https://github.com/abandonware/noble/issues/2

**Comment by rzr** [comment]
Change set released in:
https://www.npmjs.com/package/@abandonware/noble/v/1.9.2-6

Close:
https://github.com/abandonware/noble/issues/44

**Comment by rzr** [comment]
I just attempted to upstream one of your untracked change in this PR:
https://github.com/noble/noble/pull/926

**Pull request merged**